montecarlo.pl

```diff
@@ -1,118 +1,114 @@
+		take_samples_fixed(Query, Count, SamplerOpts, Probability)
+	),
 	sampler:unload_program.
 
 /**
- * montecarlo(+File:file, :Query:atom, +SamplingMethod:compound, -Propability:float) is det
- * 
- * The predicate assumes that File designates a PLP object program, which is consulted and then transformed into an equivalent standard prolog program.
- * The predicate then samples Query a number of times until the confidence (as detailed in Riguzzi 2013, p. 6) reaches a certain
- * threshold (0.02 by default) and returns the Probability of the query being true.
- * The sampling method used by this predicate can be configured via SamplingMethod by either passing the constant `standard` or `(gibbs, BlockSize)`.
- */
-montecarlo(File, Query, SamplingMethod, Probability) :-
-	montecarlo(File, Query, 0.02, SamplingMethod, Probability).
-
-/**
- * montecarlo(+File:file, :Query:atom, +Threshold:float, +SamplingMethod:atom, -Propability:float) is det
- * 
- * The predicate assumes that File designates a PLP object program, which is consulted and then transformed into an equivalent standard prolog program.
- * The predicate samples Query a number of times until the confidence (as detailed in Riguzzi 2013, p. 6) reaches the given Threshold.
- * The sampling method used by this predicate can be configured via SamplingMethod by either passing the constant `standard` or `(gibbs, BlockSize)`.
+ * montecarlo(+File:file, :Query:atom, -Probability:float) is det
+ *
+ * Alias of montecarlo/3 with empty options.
  */
-montecarlo(File, Query, Threshold, SamplingMethod, Probability) :-
-	montecarlo(File, Query, Threshold, 500, SamplingMethod, Probability).
-
-/**
- * montecarlo(+File:file, :Query:atom, +Threshold:float, +BatchSize: int, +SamplingMethod:atom, -Propability:float) is det
- * 
- * The predicate samples Query in batch sizes of BatchSize until the confidence (as detailed in Riguzzi 2013, p. 6) reaches the given Threshold.
- * The sampling method used by this predicate can be configured via SamplingMethod by either passing the constant `standard` or `(gibbs, BlockSize)`.
- */
-montecarlo(File, Query, Threshold, BatchSize, SamplingMethod, Probability) :-
-	resolve_sampling_method(SamplingMethod, SampleVia),
-	write('Performing sampling via: '), writeln(SampleVia),
-	sampler:load_program(File),
-	take_samples(Query, Threshold, BatchSize, Probability, SampleVia),
-	sampler:unload_program.
-
-take_samples(Query, Threshold, BatchSize, Probability, SampleVia) :-
-	take_samples(Query, Threshold, BatchSize, 0, 0, Probability, SampleVia).
-take_samples(Query, Threshold, BatchSize, CurrSamples, CurrSuccesses, Probability, SampleVia) :-
-	sample_batch(Query, Successes, BatchSize, SampleVia),
-	write(Successes), writeln(' samples succeeded.'),
+montecarlo(File, Query, Probability) :-
+	montecarlo(File, Query, Probability, []).
+
+resolve_sampler(Params, Sampler) :-
+	(is_list(Params) -> fail; resolve_sampler([Params], Sampler)).
```

Comment:
Aus der Zeile werd ich irgendwie nicht schlau, ist Params keine Liste von Parametern?

Reply:
@lukasrieger Das ist zur convenience damit du statt ner Liste auch nur den ersten parameter ohne in ne liste zu wrappen angeben kannst.

also es geht zb statt `sampler([standard])` auch `sampler(standard)`. KA ob man das irgendwie schöner (mit pattern matching über den anderen predicates) machen kann, ich habs nur so hinbekommen

Reply:
Ah, alles klar macht Sinn!